src/dmd/cppmangle.d

```diff
@@ -116,22 +179,7 @@ private final class CppMangleVisitor : Visitor
             /* Sequence is S_, S0_, .., S9_, SA_, ..., SZ_, S10_, ...
              */
             buf.writeByte('S');
-            if (i)
-            {
-                // Write <seq-id> to buf
-                void write_seq_id(size_t i)
-                {
-                    if (i >= 36)
-                    {
-                        write_seq_id(i / 36);
-                        i %= 36;
-                    }
-                    i += (i < 10) ? '0' : 'A' - 10;
-                    buf.writeByte(cast(char)i);
-                }
-
-                write_seq_id(i - 1);
-            }
+            writeSequenceFromIndex(i);
```

Comment:
I don't see a purpose to making this reusable, because it isn't used anywhere else. The point of nested functions like this is to make it clear it isn't used elsewhere. This refactoring makes it harder to understand, not simpler, because the logic is spread out.

Reply:
As mentioned in the commit message, this will be reused by the fix to https://issues.dlang.org/show_bug.cgi?id=16479 . Template parameter substitution use the same base36 index as "regular" substitution.